agent.py

```
@@ -147,9 +158,35 @@ async def on_participant_connected(self):
+    async def on_conversation_item_added(self, event: ConversationItemAddedEvent):
+        await super().on_conversation_item_added(event)
+        if event.item.role == "user":
+            self.conversation.append(f"Student: {event.item.content[0]}")
+        elif event.item.role == "assistant":
+            self.conversation.append(f"Kai: {event.item.content[0]}")
+
+    async def on_participant_disconnected(self):
+        await super().on_participant_disconnected()
+        with open(self.file_name, "a") as out:
+            out.write(json.dumps({"conversation": self.conversation}) + "\n")
+        await gpt.files.create(file=open(self.file_name, "rb"), purpose="evals")
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)